source/tests/system/nidaqmx_driver_api_tests.cpp

```diff
@@ -2152,15 +2319,109 @@ TEST_F(NiDAQmxDriverApiTests, WriteAnalogWaveforms_ZeroSamples_Succeeds)
   std::vector<std::vector<double>> waveform_data(1);
   waveform_data[0] = {};  // Empty waveform
   
+  ::grpc::ClientContext context;
   WriteAnalogWaveformsResponse write_response;
-  auto write_status = write_analog_waveforms(waveform_data, false, 10.0, write_response);
+  auto write_status = write_analog_waveforms(context, waveform_data, false, 10.0, write_response);
   
   EXPECT_SUCCESS(write_status, write_response);
   EXPECT_EQ(write_response.samps_per_chan_written(), 0);
+  EXPECT_EQ(get_from_trailing_metadata(context, "ni-samps-per-chan-written"), std::to_string(0));
+  
+  stop_task();
+}
+
+TEST_F(NiDAQmxDriverApiTests, ReadAnalogWaveforms_InvalidNumSampsPerChan_ReturnsErrorAndZeroMetadata)
+{
+  const auto INVALID_NUM_SAMPLES = -2;
+  const auto TIMEOUT = 10.0;
+  CreateAIVoltageChanResponse create_channel_response;
+  auto create_channel_status = create_ai_voltage_chan("gRPCSystemTestDAQ/ai0", "ai0", -1.0, 1.0, create_channel_response);
+  EXPECT_SUCCESS(create_channel_status, create_channel_response);
+
+  start_task();
+  
+  EXPECT_THROW({
+    ::grpc::ClientContext context;
+    ReadAnalogWaveformsResponse read_response;
+    auto read_status = read_analog_waveforms(context, INVALID_NUM_SAMPLES, TIMEOUT, WaveformAttributeMode::WAVEFORM_ATTRIBUTE_MODE_NONE, read_response);
+    
+    EXPECT_EQ(get_from_trailing_metadata(context, "ni-samps-per-chan-read"), "0");
```

Comment:
This EXPECT_EQ doesn't execute because the function throws before executing it. Move it outside of the EXPECT_THROW block.